### PR TITLE
Chore – Migrate to Solr 8.7 and  Postgres 11

### PR DESCRIPTION
**Pull request by alfonsomunozpomer**
Dear reviewer:
Don’t be scared by the 247 files chaned. Most of those are because of the removal of the `packages` directory. With Solr 8 and the ability of doing POST requests in SolrJ out-of-the-box, we don’t need to fork and maintain our version of the libraries. You’ll see here the new Solr 8 dependencies and adding HTTP headers with authentication, as well as the new `solr.hosts` and `zk.hosts` properties, which replace the old `solr.host` and `zk.host`, since [the old `SolrCloud` builders have been deprecated](https://solr.apache.org/docs/8_7_0/solr-solrj/org/apache/solr/client/solrj/impl/CloudSolrClient.Builder.html#withSolrUrl-java.util.Collection-).

However, the biggest change is the new `README.md` and the set of scripts to prepare your development environment.

Please, follow the documentation provided and run the scripts to verify everything works as expected. Please, don’t hesitate to suggest clarifications/rewording in `README.md` if anything is unclear.

Related PR in `atlas-web-core`: https://github.com/ebi-gene-expression-group/atlas-web-core/pull/115

Godspeed!

### CODE REVIEW

File: Jenkinsfile
Comment (ke4) [inline Jenkinsfile]: Is this accidentally commented out or deliberately?
If we don't need it then we should delete this line.
We still have it in the git history if we need it later.

File: Jenkinsfile
Comment (ke4) [inline Jenkinsfile]: Shall we put back `main` instead of `chore/solr8`, please?

File: Jenkinsfile
Comment (ke4) [inline Jenkinsfile:175]: Please remove this line if we don't need it.
Also, line 179.
Thanks

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java
Comment (ke4) [inline app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java:170]: Is there a ticket for this in our backlog?

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java
Comment (ke4) [inline app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java:178]: We should remove these 4 lines as they are commented out.

File: debug-single-test.sh
Comment (ke4) [inline debug-single-test.sh:18]: Do we need this again?
We set this in line 2.

File: docker/docker-compose-gradle.yml
Comment (ke4) [inline docker/docker-compose-gradle.yml]: We could just add the next line without commented out. I think that would be more helpful. Somebody running those tests would not have to search for this to remove the comment sign before the next line.

File: docker/prepare-dev-environment/gradle-ro-dep-cache/Dockerfile
Comment (ke4) [inline docker/prepare-dev-environment/gradle-ro-dep-cache/Dockerfile]: Shall we change the name of the branch from `chore/solr8` to `main` as that branch might not exist after merging and deleting that branch?

File: docker/prepare-dev-environment/gradle-ro-dep-cache/run.sh
Comment (ke4) [inline docker/prepare-dev-environment/gradle-ro-dep-cache/run.sh:7]: Could we remove these 3 lines, please? Tx

File: docker/prepare-dev-environment/postgres/run.sh
Comment (ke4) [inline docker/prepare-dev-environment/postgres/run.sh:7]: Do we need these lines? and also from lines 14-16? Tx

File: docker/prepare-dev-environment/solr/docker-entrypoint.sh
Comment (ke4) [inline docker/prepare-dev-environment/solr/docker-entrypoint.sh:53]: 😅 So, that's why it runs at least 3+ h 😊

File: docker/prepare-dev-environment/solr/run.sh
Comment (ke4) [inline docker/prepare-dev-environment/solr/run.sh:5]: Could we remove these lines, please?
5-6
8-11
14-15
Tx

File: docker/prepare-dev-environment/volumes/run.sh
Comment (ke4) [inline docker/prepare-dev-environment/volumes/run.sh:5]: Could we remove these lines, please?
5
7-10
12-14
Tx

File: docker/prepare-dev-environment/volumes/run.sh
Comment (ke4) [inline docker/prepare-dev-environment/volumes/run.sh:78]: Shall we move these variables to `dev.env` file?

File: execute-all-tests.sh
Comment (ke4) [inline execute-all-tests.sh:17]: We already set this on line 2.

File: execute-single-test.sh
Comment (ke4) [inline execute-single-test.sh:18]: We already set this on line 2.

File: stop-and-remove-containers.sh
Comment (ke4) [inline stop-and-remove-containers.sh]:
```suggestion
scxa-postgres scxa-postgres-test scxa-flyway-test \
```

File: stop-and-remove-containers.sh
Comment (ke4) [inline stop-and-remove-containers.sh:10]:
```suggestion
scxa-tomcat scxa-gradle
```

File: stop-and-remove-containers.sh
Comment (ke4) [inline stop-and-remove-containers.sh]:
```suggestion
scxa-postgres  scxa-postgres-test scxa-flyway-test \
```

File: stop-and-remove-containers.sh
Comment (ke4) [inline stop-and-remove-containers.sh:16]:
```suggestion
scxa-tomcat  scxa-gradle
```

File: README.md
Comment (ke4) [inline README.md]:
```suggestion
Just add the necessary species names and experiment accessions in the `test-dev.env` file and rebuild the development 
```

File: README.md
Comment (ke4) [inline README.md]:
```suggestion
The second argument is optional and can be used to specify a database version. As before, if unspecified it defaults to
```

File: README.md
Comment (ke4) [inline README.md]:
```suggestion
./gradlew clean :app:war
```

File: README.md
Comment (ke4) [inline README.md]:
```suggestion
You should now have the file `webapps/gxa#sc.war`. Because the directory `webapps` is bind-mounted in Tomcat’s 
```

File: README.md
Comment (ke4) [inline README.md]:
```suggestion
If you get any `war` redeployment issues or want to start again freshly, stop all the containers using this:
```

**Review by ke4 — CHANGES_REQUESTED** [review]
Hey @alfonsomunozpomer,
It is very great work. I added some minor comments.
Could you please review them?
Many thanks 👍

File: Jenkinsfile
Comment (alfonsomunozpomer) [reply Jenkinsfile]: Well-spotted! I had forgotten about this. Let’s see what happens on Jenkins end now with the GKE cluster...

File: Jenkinsfile
Comment (alfonsomunozpomer) [reply Jenkinsfile:175]: I’m hesitant about removing stuff related to integration tests of `atlas-web-core` to have a constant reminder that we should enable them as soon as we have a bulk dev/test environment.

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java
Comment (alfonsomunozpomer) [reply app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java:170]: This is a comment about fixtures. The new anndata fixtures solve this issue, it’s removed there.

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java
Comment (alfonsomunozpomer) [reply app/src/test/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentServiceIT.java:178]: They’ve been re-enabled in the anndata branch.

File: stop-and-remove-containers.sh
Comment (ke4) [reply stop-and-remove-containers.sh]: I believe the name of the postgres container should be without `test`:

`scxa-postgres scxa-flyway-test \`

File: stop-and-remove-containers.sh
Comment (ke4) [reply stop-and-remove-containers.sh:10]: Here is the same as above:
the name of the postgres container should be without test:

`scxa-postgres scxa-flyway-test \`

File: debug-single-test.sh
Comment (alfonsomunozpomer) [reply debug-single-test.sh:18]: Yes. One is for the script in the host machine, while this one is for the Bash script in the Docker container.

```
$ docker run --rm ubuntu:jammy bash -c "
foobar
echo 'Hello!'
"
bash: line 2: foobar: command not found
Hello!
### End of output
```

Versus:
```
$ docker run --rm ubuntu:jammy bash -c "
set -e
foobar
echo 'Hello!'
"
bash: line 3: foobar: command not found
### End of output
```

Comment (ke4) [reply debug-single-test.sh:18]: ok, tx

File: docker/docker-compose-gradle.yml
Comment (alfonsomunozpomer) [reply docker/docker-compose-gradle.yml]: I created a story to add `atlas-web-core` tests in the Gradle container.

File: execute-single-test.sh
Comment (alfonsomunozpomer) [reply execute-single-test.sh:18]: Yes. One is for the script in the host machine, while this one is for the Bash script in the Docker container.

```
$ docker run --rm ubuntu:jammy bash -c "
foobar
echo 'Hello!'
"
bash: line 2: foobar: command not found
Hello!
### End of output
```

Versus:
```
$ docker run --rm ubuntu:jammy bash -c "
set -e
foobar
echo 'Hello!'
"
bash: line 3: foobar: command not found
### End of output
```

File: execute-all-tests.sh
Comment (alfonsomunozpomer) [reply execute-all-tests.sh:17]: Yes. One is for the script in the host machine, while this one is for the Bash script in the Docker container.

```
$ docker run --rm ubuntu:jammy bash -c "
foobar
echo 'Hello!'
"
bash: line 2: foobar: command not found
Hello!
### End of output
```

Versus:
```
$ docker run --rm ubuntu:jammy bash -c "
set -e
foobar
echo 'Hello!'
"
bash: line 3: foobar: command not found
### End of output
```

File: docker/prepare-dev-environment/gradle-ro-dep-cache/Dockerfile
Comment (alfonsomunozpomer) [reply docker/prepare-dev-environment/gradle-ro-dep-cache/Dockerfile]: Yes, but now there’s code that is needed in this branch. I’ll do this as the last step before clicking the Merge button.

File: docker/prepare-dev-environment/gradle-ro-dep-cache/run.sh
Comment (alfonsomunozpomer) [reply docker/prepare-dev-environment/gradle-ro-dep-cache/run.sh:7]: I tried to be explicit about what we’re importing from where. If you think it’s very noisy I can remove it. I thought it’d be helpful for whomever is reading the code.

File: docker/prepare-dev-environment/volumes/run.sh
Comment (alfonsomunozpomer) [reply docker/prepare-dev-environment/volumes/run.sh:78]: The reason why it’s not part of `dev.env` is that the target directories (e.g. `/atlas-data/bioentity_properties`) is specific to the container in `volumes/Dockerfile`. We can potentially have different mappings depending on the container structure and it could get a bit confusing.

File: stop-and-remove-containers.sh
Comment (alfonsomunozpomer) [reply stop-and-remove-containers.sh]: I’ve parameterised the script.

File: stop-and-remove-containers.sh
Comment (alfonsomunozpomer) [reply stop-and-remove-containers.sh:10]: I’ve parameterised the script.

File: stop-and-remove-containers.sh
Comment (alfonsomunozpomer) [reply stop-and-remove-containers.sh]: I’ve parameterised the script.

File: docker/prepare-dev-environment/gradle-ro-dep-cache/Dockerfile
Comment (ke4) [reply docker/prepare-dev-environment/gradle-ro-dep-cache/Dockerfile]: OK, but please don't forget to do it

**Review by ke4 — DISMISSED** [review]
@alfonsomunozpomer thanks for working on this branch so hard. I think it is more than perfect now. 👍 💯

**Review by ke4 — DISMISSED** [review]

**Pull request merged**
